mod_reforged/hooks/experimental_modules/onAnySkillExecutedFully.nut

```diff
@@ -69,6 +100,40 @@ local function getSchedulerSkill()
 	return null;
 }
 
+// Returns a wrapped callback function for schedule functions e.g. ::Time.scheduleEvent. The callback
+// is wrapped to trigger the onScheduleComplete() function for the schedule.
+// Also increments the schedule Count of the relevant skill and saves the actor state when at Count 0.
+// Note: _countBump is necessary as a param because ::Tactical.switchEntities needs to increase count by 2 on every call
+// as its callback is triggered twice: once for each entity being switched.
+local function getWrapper( _caller, _func, _numArgs, _countBump = 1 )
+{
+	local schedule = ::Reforged.ScheduleSkills[_caller];
+	if (schedule.Count == 0)
+	{
+		schedule.saveStartingState();
```

Comment:
A really weird place to do it.

Reply:
I have updated the PR by moving state save/check into an agent independent of scheduled events.